flixel/system/FlxSound.hx

```diff
@@ -737,19 +732,55 @@ class FlxSound extends FlxBasic
 		return Volume;
 	}
 
-	#if (sys && openfl_legacy)
 	inline function get_pitch():Float
 	{
 		return _pitch;
 	}
 
 	function set_pitch(v:Float):Float
 	{
-		if (_channel != null)
-			_channel.pitch = v;
+		@:privateAccess
+		#if openfl_legacy
+			#if sys
+			@:privateAccess
+			if (_channel != null)
+				_channel.__source.pitch = v;
+			#else
+			FlxG.log.error("Pitch is supported only in sys systems while using openfl legacy!! switch the build into sys or upgrade your openfl!");
+			return 1;
+			#end
+		#else
+		#if flash
+		FlxG.log.error("Pitch is not supported on Flash!!!");
+		v = 1;
+		#elseif (html5 && lime_howlerjs)
+		_channel.__source.buffer.__srcHowl.rate(v);
+		#else
+		var backend = _channel.__source.__backend;
+		@:privateAccess
+		if (backend.playing && v != _pitch)
+		{
+			if (backend.timer != null)
+			{
+				backend.timer.stop();
+			}
+
+			var timeRemaining = Std.int((length - time) / v);
+			if (timeRemaining > 0)
+			{
+				backend.timer = new haxe.Timer(timeRemaining);
+				backend.timer.run = backend.timer_onRun;
+			}
+		}
```

Comment:
I'm not sure what's being done here, can you add comments explaining?

Reply:
Basically, in a song (at least how lime treats it on cpp) there's a timer that if it finishes, it calls the finished stuff.
if you changed the pitch without changing the timer of the song, you would have the pitch, but the song would "end" too late or too early depending of the pitch cos the time would be on pitch 1.
basically what it does is calling internal lime stuff so that it would be available not only on lime 3.1.0 (or idk what version is gonna be) but other versions too.

Reply:
I think this might be a bit better off being more integrated with OpenFL, and then flixel simply calls to set the new pitch, a bit similarly how it'd just simply set to change the pitch with the old pitch logic, although of course this time, should just need something like george suggests, `#if !flash` or something like that :)

Reply:
> I think this might be a bit better off being more integrated with OpenFL, and then flixel simply calls to set the new pitch, a bit similarly how it'd just simply set to change the pitch with the old pitch logic, although of course this time, should just need something like george suggests, `#if !flash` or something like that :)

this would be less code and would make this 100% better, but the thing is, idk when lime is gonna release, and if I just add that up, this would a feature that would only work for a git version of lime, which normally is broken cos it's on dev.
if the next lime version is releasing, I would totally do that but if you do that rn, it wouldn't be a feature unless lime releases the next version and it could be days/weeks/months till it releases.
unless George doesn't care too much about it oc

Reply:
keep in mind the next flixel release isn't happening soon (5.0.0) so might as well wait for lime, right?

Reply:
looks like lime is planning to release, soon
https://community.openfl.org/t/lime-and-openfl-update/13539

Reply:
I this is resolved 100% uncertain